src/components/Public/PortalSubmission.js

```diff
@@ -23,26 +23,125 @@ const PortalSubmission = () => {
     const [videoName, setVideoName] = useState("");
 
     const [submission, setSubmission] = useState(false);
+    const [validEmail, setValidEmail] = useState(false);
 
     // [Title to be shown, id of title for database]
     const requiredFields = [
-        ["First name", "firstName"],
-        ["Last name", "lastName"],
-        ["BU Email address", "email"],
-        ["Phone number", "phone"],
-        ["Expected year of graduation", "gradYear"],
-        ["Major", "major"]
+        {
+            label: 'First name',
+            name: 'firstName',
+            type: 'text'
+        },
+        {
+            label: 'Last name',
+            name: 'lastName',
+            type: 'text'
+        },
+        {
+            label: 'Major',
+            name: 'major',
+            type: 'text'
+        },
+        {
+            label: 'Expected graduation date',
+            name: 'gradYear',
+            type: 'date'
+        },
+        // {
+        //     label: 'Email address',
+        //     name: 'email',
+        //     type: 'email'
+        // },
+        // {
+        //     label: 'Phone number',
+        //     name: 'phone',
+        //     type: 'tel'
+        // },
     ]
 
     const optionalFields = [
-        ["LinkedIn Profile", "linkedin"],
-        ["Website/Portfolio", "website"]
+        {
+            label: 'LinkedIn Profile',
+            name: 'linkedin',
+            type: 'url'
+        },
+        {
+            label: 'Website / Portfolio',
+            name: 'website',
+            type: 'url'
+        }
     ]
 
     const selectFields = [
         ["How did you hear about PCT?", "marketing"]
     ]
 
+    function validateEmail(event) {
+        const regexp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        console.log(regexp.test(event.target.value))
+        setValidEmail(regexp.test(event.target.value))
+    }
+
+    function emailForm() {
```

Comment:
I'm wondering if creating separate components would be better for emailForm and phoneForm (this is fine but just something to consider!)